ckan/logic/validators.py

```diff
@@ -348,16 +349,16 @@ def package_name_validator(key, data, errors, context):
     session = context['session']
     package = context.get('package')
 
-    query = session.query(model.Package.name).filter_by(name=data[key])
+    query = session.query(model.Package.state).filter_by(name=data[key])
     if package:
         package_id = package.id
     else:
         package_id = data.get(key[:-1] + ('id',))
     if package_id and package_id is not missing:
         query = query.filter(model.Package.id <> package_id)
     result = query.first()
-    if result:
-        errors[key].append(_('That URL is already in use.'))
+    if result and result.state != State.DELETED:
+            errors[key].append(_('That URL is already in use.'))
```

Comment:
this is good but the indentation is off :confused: wonder why our code style checker didn't catch that